server/src/main/java/org/elasticsearch/cluster/RestoreInProgress.java

```diff
@@ -382,6 +383,11 @@ public String getWriteableName() {
         return TYPE;
     }
 
+    @Override
+    public Version getMinimalSupportedVersion() {
+        return Version.V_5_0_0;
```

Comment:
It is unfortunate that this has to be littered everywhere and anyway it will become a lie when we remove the 5.x constants from the codebase on some future cleanup. Why not use `Version#minimumCompatibilityVersion` as a default implementation on the interface?

Reply:
I felt that people should really think about these things and not just rely on defaults. Another aspect is that this change currently means it is littered everywhere in this PR but when people introduce a new `VersionedNamedWritable` the have to set something can't use a default. When 5.x is phased out we can replace it with `Version#minimumCompatibilityVersion` (I can do so now if you want) but I think it should be done on a case by case basis. Defaults are dangerous here imo.

Reply:
Fair enough. Let's go with `Version#minimumCompatibilityVersion` everywhere now though.

Reply:
Sure.

Reply:
I replaced all occurrences of V5_0_0. Let me know if you want the other 5.x replaced as well.

Reply:
Yeah, let's get those here too.